### DISCUSSION TITLE
Transparent Image Background Doesnt remove all white

### DISCUSSION THREAD

**Discussion by aaronctmd** [ImageMagick/ImageMagick#4644] [Digital Image Processing]
Hello I am using code 

for %F in (*.jpg) do magick convert %~nF.jpg -transparent white %~nF.png  

To convert image file to .png and also remove the background .

On most of the images there is still white around the image .  Can i fix this with -fuzz and if so how? 

Where would i input it in this code: 

for %F in (*.jpg) do magick convert %~nF.jpg -transparent white %~nF.png  

see example of image attached
![newimage](https://user-images.githubusercontent.com/96889493/147802891-59db1026-c114-4819-940f-816136f52994.png)

**Comment by fmw42** [comment]
You can try (note use magick not magick convert)

`for %F in (*.jpg) do magick %~nF.jpg -fuzz XX% -transparent white %~nF.png
`

where XX% is in range 0% to 100%. Start with 10% and increase slowly until you get what you want. If in a .bat file, then double the %.

If that does not work, then anti-aliasing with a blur might be needed. But that is more complex.

**Reply by aaronctmd** [reply]
@fmw42  Helllo again. that worked but it made some of the images distort see example pi
![13591_4](https://user-images.githubusercontent.com/96889493/147806024-378f98b6-b6c7-456f-9cdc-204fd4febf15.png)
c

**Comment by fmw42** [comment]
Post one of your input images.

**Reply by fmw42** [reply]
Your drink cup image above has no transparency. Why is that?  Are you doing more than just the -transparent command you showed?

**Reply by fmw42** [reply]
Please always post your full set of commands and the input and output commands for the commands you post.

**Comment by Jmuccigr** [comment]
Just noting that `mogrify -format png filename.jpg` will rename the file to `filename.png` all on its own. It will also handle the files one at a time, so you won't run into any space issues if there are a lot of them.

So in bash:

`for f in {1..10}; do mogrify -format png -fuzz "$f"% -transparent white *.jpg; done`
